gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/push/ValidPushMessage.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.push;
+
+import jakarta.validation.Constraint;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * A class-level validation constraint for PubSub push messages.
+ *
+ * @author Jeremy Grelle
+ * @since 5.4.0
+ */
+@Retention(RUNTIME)
+@Target(TYPE)
+@Constraint(validatedBy = { PushMessageValidator.class })
+public @interface ValidPushMessage {
```

Comment:
Define the validation annotation with the standard structure. See: https://github.com/micronaut-projects/micronaut-email/blob/master/email/src/main/java/io/micronaut/email/validation/AnyRecipient.java

Reply:
https://github.com/micronaut-projects/micronaut-gcp/pull/1012/commits/51a0a5a726217308e3804b4bc68f09da021c56ed